### PR TITLE
Use a shared ipam namespace by default.

### PR DESCRIPTION
**Pull request by chirino**
* Also switch to the CG-NAT cird 100.64.0.0/10
* Set org `private_cidr=true` to use a non-shared ipam namespace for the org.

### CODE REVIEW

**Comment by russellb** [comment]
blocked on merging design proposal in #998

**Comment by russellb** [comment]
s/cider/cidr/ in the commit message :)

**Comment by nerdalert** [comment]
Ran it through QA a couple of times 🟢 
https://github.com/nexodus-io/nexodus/actions/runs/4951380705
https://github.com/nexodus-io/nexodus/actions/runs/4952463061

**Comment by russellb** [comment]
I posted #1168 for a seg fault I saw in the last e2e test run on here.

**Review by russellb — CHANGES_REQUESTED** [review]
fairly minor comments. This was much simpler than I expected.

**Review by nerdalert — APPROVED** [review]
LGTM, leaving the ability to spin up a custom CIDR addresses my only hesitation with `--cidr 100.100.0.0/16 --cidr-v6 200::/32`.

That coupled with the ability for a user to choose what org a device joins gives the enterprise deployment freedom to customize divisions that aren't SaaS deployments and would prefer sequential addressing for ease in policy etc.

**Comment by chirino** [comment]
> LGTM, leaving the ability to spin up a custom CIDR addresses my only hesitation with --cidr 100.100.0.0/16 --cidr-v6 200::/32.

Not sure what you mean. This pull request supports doing that.

**Comment by russellb** [comment]
> > LGTM, leaving the ability to spin up a custom CIDR addresses my only hesitation with --cidr 100.100.0.0/16 --cidr-v6 200::/32.
> 
> Not sure what you mean. This pull request supports doing that.

I was wondering if the response was just misworded a bit. I interpreted it as "this looks good. Maintaining the ability use custom CIDRs resolved my only hesitation."

**Comment by nerdalert** [comment]
> > LGTM, leaving the ability to spin up a custom CIDR addresses my only hesitation with --cidr 100.100.0.0/16 --cidr-v6 200::/32.
> 
> Not sure what you mean. This pull request supports doing that.

@russellb @chirino Sorry, meant my only reservation was addressed with that. I'm a go here!

File: internal/database/database.go [hunk 1]
```diff
@@ -3,6 +3,11 @@ package database
 import (
 	"context"
 	"fmt"
+	"github.com/nexodus-io/nexodus/internal/database/migration_20230409_0000"
+	"github.com/nexodus-io/nexodus/internal/database/migration_20230411_0000"
+	"github.com/nexodus-io/nexodus/internal/database/migration_20230412_0000"
+	"github.com/nexodus-io/nexodus/internal/database/migration_20230413_0000"
+	"github.com/nexodus-io/nexodus/internal/database/migration_20230509_0000"
```
[the remark this answers is not in the document]
Comment (chirino) [reply internal/database/database.go:10]: Yeah... Maybe we should all switch to InteliJ 😆

File: integration-tests/security_group_test.go [hunk 2]
```diff
@@ -231,12 +231,12 @@ func TestSecurityGroups(t *testing.T) {
...
 	// create the new inbound and outbound rules
 	inboundRules = []public.ModelsSecurityRule{
-		helper.createSecurityRule("icmpv4", "0", "0", []string{"100.100.0.1-100.100.0.50"}),
+		helper.createSecurityRule("icmpv4", "0", "0", []string{"100.64.0.1-100.127.0.50"}),
 		helper.createSecurityRule("icmpv6", "0", "0", []string{""}),
 	}
 	outboundRules = []public.ModelsSecurityRule{
 		helper.createSecurityRule("icmpv4", "0", "0", []string{""}),
-		helper.createSecurityRule("icmpv6", "0", "0", []string{"200::1-200::50"}),
+		helper.createSecurityRule("icmpv6", "0", "0", []string{"200::1-200:1::50"}),
```
[the remark this answers is not in the document]
Comment (chirino) [reply integration-tests/security_group_test.go:239]: I'll change it... but I wonder what we should be seeing in tests that configure bad rules like this...

File: internal/handlers/organization.go [hunk 3]
```diff
@@ -67,12 +69,16 @@ func (api *API) CreateOrganization(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, models.NewBadPayloadError())
 		return
 	}
+	if !request.PrivateCidr {
+		request.IpCidr = defaultIPAMv4Cidr
+		request.IpCidrV6 = defaultIPAMv6Cidr
+	}
```
[the remark this answers is not in the document]
Comment (chirino) [reply internal/handlers/organization.go:75]: ok

File: integration-tests/security_group_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (russellb) [reply integration-tests/security_group_test.go:239]: The rule isn't necessarily invalid. You can configure any IP range as long as it's a valid IP range. It's just that I knew this extended beyond the range of what nexodus would ever allocate.

File: internal/database/database.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (russellb) [reply internal/database/database.go:10]: If this change was InteliJ trying to be smart, then I'll definitely pass. 😆

**Comment by russellb** [comment]
@chirino looks OK, but unit tests are busted

**Pull request merged**